### PR TITLE
Fix Open Trades scrollbar

### PR DESCRIPTION
**Pull request by preland**
This PR wraps the pre-existing footer in a ScrollPane, which gives full scrolling functionality to the area.

### CODE REVIEW

**Comment by preland** [comment]
(fixes #677)

**Comment by woodser** [comment]
I'm seeing the scrollbar working in the bottom panel, but it's visible even when it shouldn't be necessary. Here's a screenshot with a short payment method and full screen resolution:

image

It appears extra padding was added to the top and bottom of the panel or a fixed height and the scrollbar is always visible.

**Comment by preland** [comment]
The newest commit should fix that issue

**Comment by woodser** [comment]
Thanks. Please remove the unused import to fix the build.

**Comment by preland** [comment]
👍

File: Makefile [hunk 1]
```diff
@@ -74,7 +74,7 @@ monerod1-local:
 		--add-exclusive-node 127.0.0.1:48080 \
 		--add-exclusive-node 127.0.0.1:58080 \
 		--rpc-access-control-origins http://localhost:8080 \
-		--fixed-difficulty 500 \
+		--fixed-difficulty 1 \
```
Comment (woodser) [inline Makefile:77]: Remove change

File: Makefile [hunk 2]
```diff
@@ -92,7 +92,7 @@ monerod2-local:
 		--add-exclusive-node 127.0.0.1:28080 \
 		--add-exclusive-node 127.0.0.1:58080 \
 		--rpc-access-control-origins http://localhost:8080 \
-		--fixed-difficulty 500 \
+		--fixed-difficulty 1 \
```
Comment (woodser) [inline Makefile:95]: Remove

File: desktop/src/main/java/haveno/desktop/main/portfolio/pendingtrades/PendingTradesView.java [hunk 3]
```diff
@@ -99,6 +99,8 @@
 
 @FxmlView
 public class PendingTradesView extends ActivatableViewAndModel<VBox, PendingTradesViewModel> {
+    private ScrollPane subViewScroll;//this is up here because of codacy
```
Comment (woodser) [inline desktop/src/main/java/haveno/desktop/main/portfolio/pendingtrades/PendingTradesView.java:102]: Please group with other fields, remove comment, and ignore codacy's complaints.

**Review by woodser — COMMENTED** [review]
Nice, it's working. Just a few cleanup comments.

**Comment by woodser** [comment]
Also please squash to a single commit.

**Comment by preland** [comment]
extraneous changes were removed; PR is now squashed

**Comment by woodser** [comment]
Thanks, please also squash the commit message. A single line will do.

**Comment by preland** [comment]
The commit message is now shortened

**Comment by woodser** [comment]
I just realized this PR increases the height of the trade list substantially:

image

This height should stay the same (or increase very slightly, because it's pretty short). That will make the scrollbar unnecessary and hidden in most cases.

**Comment by woodser** [comment]
It should match the original behavior as best as possible, which is optimized for very few active trades (most common) and maximizes the remaining space.

**Comment by preland** [comment]
Review the latest commit and see if that looks better

**Comment by woodser** [comment]
The height of the trades list is still a bit taller than before.

Before:

image

After:

image

Since most people are taking one or two trades at a time, we should keep it shorter to allocate that space to the controls. About two trades should fit snugly.

**Comment by preland** [comment]
I see what you mean; I think this is what you want? (the commit is amended)

**Comment by woodser** [comment]
I'm seeing that the text areas for Pay by Mail are too large and cause scrolling with this PR. Probably it's the same for other payment methods with text areas.

Before:

image


After:

image

**Comment by preland** [comment]
That commit should fix that issue; I also went through the forms to look for and change the relevant fields. I also noticed that the left panel was sized incorrectly in your screenshot, so I fixed the width of it

**Comment by woodser** [comment]
This PR is introducing horizontal scrolling due to the text areas being too wide:

Before:

image

After:

image

**Comment by woodser** [comment]
I was thinking, would it be easier if the whole panel was in a scroll pane, instead of only the trade content?

**Comment by preland** [comment]
I had tried that earlier, but the main issue was that the subview area refused to squash, and it wouldn’t grow past a certain size (which would cause the weird issue of it showing an unscrollable scrollbar while still having content off-screen)

This commit should fix the issue; although I will admit I had difficulty getting the issue to reproduce consistently on my system

**Comment by woodser** [comment]
I'm still seeing quite a bit of width added to the payment details. This is from a fresh build:

image

Hoping it can be removed?

**Comment by preland** [comment]
This should do it; I had this cap on initially for the commit, but removed it since the view wasn't increasing past a certain point without it

**Comment by woodser** [comment]
I'm still seeing the extra width added. To verify development, hopefully you are able to see the same from a fresh build with Pay by Mail?

image

**Comment by preland** [comment]
I haven't been able to completely reproduce what you've been seeing; I can give more details about exact size differences later; I've been using different resolutions from 720p to 2160p, and the size of the area remains consistent. Could you give me some more info about the resolution that this is occurring on, as well as how the text scales at different resolutions for you?

**Comment by woodser** [comment]
It appears mostly the same for me at any resolution. This is with the application's default size, so it might be necessary to start from a fresh build to see it as I see it:

1168 x 755:

image

2056 x 1329:

image

**Comment by preland** [comment]
I’ll see if I can reproduce it

**Comment by preland** [comment]
Hopefully this one fixes it

**Comment by woodser** [comment]
I'm seeing that the width is fixed now so that's good.

However the width of the payment details has been squashed for many payment methods:

Before:

image

After:

image

And it introduces a horizontal scrollbar when the app has default dimensions for at least Pay by Mail:

image

We want to preserve the current UI with the addition of the scrollbar. :)

**Comment by woodser** [comment]
Fixed with https://github.com/haveno-dex/haveno/pull/773

**Pull request closed without merging**